src/main/com/fulcrologic/fulcro/algorithms/normalized_state_helpers.cljc

```diff
@@ -0,0 +1,309 @@
+                                      ps
+                                      m))]
+                            (filter #(< (count %) 4)
+                                    (paths* () [] state-map)))
+
+         ident-specific-paths (fn [state ident]
+                                (filter (fn [a-path]
+                                          (let [vl (clojure.core/get-in state a-path)]
+                                            (if (coll? vl)
+                                              (or
+                                                (some #{ident} vl)
+                                                (= ident vl)))))
+                                        normalized-paths))
+
+         remove-ident-at-path (fn [state a-normalized-path ident]
+                                (let [v (clojure.core/get-in state a-normalized-path)]
+                                  (if (coll? v)
+                                    (cond
+                                      (= v ident) (dissoc-in state a-normalized-path)
+                                      (every? eql/ident? v) (merge/remove-ident* state ident a-normalized-path)
+                                      :else state)
+                                    state)))
+
+
+         remove-ident-from-tables (fn [state ident]
+                                    (reduce #(remove-ident-at-path %1 %2 ident)
+                                            state
+                                            (ident-specific-paths state ident)))
+
+         state-without-entity (->
+                                ;; remove pointers to the entity
+                                (remove-ident-from-tables state-map ident)
+                                ;; remove the top-level entity
+                                (dissoc-in ident))
+
+         target-entity (get-in state-map ident)
+
+         ;; Computed set of all affected entities when cascade option is provided
+         cascaded-idents (let [table-key (first ident)]
+                           (map
+                             (fn [entity-field]
+                               (clojure.core/get-in state-map
+                                                    (conj [table-key (table-key target-entity)] entity-field)))
+                             (set/intersection
+                               cascade
+                               (set (keys target-entity)))))
+
+         final-state (reduce
+                       (fn [state edge]
+                         (if (every? eql/ident? edge)
+                           (reduce (fn [new-state ident] (remove-entity* new-state ident cascade)) state edge)
+                           (remove-entity* state edge cascade)))
+                       state-without-entity
+                       cascaded-idents)]
+
+     final-state)))
+
+
+(>defn remove-edge*
+  "Remove the given edge at the given path. Also scans all tables and removes any to-one or to-many idents that are
```

Comment:
Now that I'm looking at this one I'm not loving it.

I think this needs additional option(s).  The use-case in applications is that you're writing a mutation where
you know the edge name, and you'd like to unlink it from the graph; however, it is often the case that you do *not* want to actually cause the removal of the entity itself (that's remove-entity*).  For convenience, this function *becomes* a convenience interface for `remove-entity*`.  Technically what is at the edge is one of 3 things:

1. A map or scalar: ignore the request
2. A single ident: Two choices. 1. Remove the edge (simple dissoc). 2. Call `remove-entity*` with that ident.
3. Many idents: Reduction using rule in (2).

Thus, being able to tell it which behavior (dissoc vs. remove-entity), and *then if* it is remove entity, where to cascade. 

In real mutations you're often going to want to use `ref` (remove edge from "this"):

```
(-> s
  (remove-edge* (conj ref :person/children)))
```

or perhaps some top-level edge `(remove-edge* state-map [:root-key])`.

Cascading/recursion is rarer in these cases.

Perhaps this?

`(remove-edge* s path)` - pretty much dissoc-in. No entities are removed, just the edge.
`(remove-edge* s path #{})` - Removes the entity (and all refs to it), but since cascade set is empty goes no further.
`(remove-edge* s path #{:edge})` - Follows the given edges as well as removing the entity.

Thoughts?

Reply:
one question, since those are public functions, why don't we remove the `*` from the end of the name? some old had because were internal, but looks to me for a public one, what you think?

Reply:
Our naming convention is a `*` suffix if it is meant to operate on the
state map (not atom)...has nothing to do with internal vs external. If we wanted to abandon that naming convention it leave us in a state where a lot of the older ones had the suffix and the newer ones do not, not to mention the stuff I've written about this in the book.  

That said, I realize that it is a non-standard naming convention. I'd be willing to consider a proposal to drop the `*`, and add aliases for all of the existing ones and dropping the `*`.

Reply:
I'm down for that (dropping `*`), better to have the ones to atom to have `!`, so we follow regular conventions for Clojure

Reply:
My experience is, of course, quite limited in contrast to both of you. The only comment I'd add is that I agree sticking to the standard clojure naming convention however, the `aliases` part gives me a hiccup for sure. It might make the API surface seem unnecessarily large which might end up confounding people for no reason.

Reply:
> `(remove-edge* s path)` - pretty much dissoc-in. No entities are removed, just the edge.

Mm, if it is a simple `dissoc-in` ( which is an internal function util as of now), why not expose it and let the users use it explicitely

Reply:
👍  about leaving `dissoc-in` public